File: StriverA2Z/C++/13_BinaryTree/flatten_BT_to_LL.cpp
#include<bits/stdc++.h>
#include "Node.h"

using namespace std;

void flatten(Node *root) {
    Node *cur = root;
    while (cur) {
        if (cur->left) {
            Node *prev = cur->left;
            while (prev->right) prev = prev->right;
            prev->right = cur->right;
            cur->right = cur->left;
            cur->left = nullptr;
        }
        cur = cur->right;
    }
}

int main() {
    return 0;
}

File: StriverA2Z/Java/BinaryTrees/FlattenBTToLL.java
public class FlattenBTToLL {
    public static void main(String[] args) {

    }

    public void flatten(Node root) {
        Node cur = root;
        while (cur != null) {
            if (cur.left != null) {
                Node prev = cur.left;
                while (prev.right != null) prev = prev.right;
                prev.right = cur.right;
                cur.right = cur.left;
                cur.left = null;
            }
            cur = cur.right;
        }
    }
}
